sycl/include/sycl/ext/intel/experimental/esimd/math.hpp

```diff
@@ -190,11 +296,10 @@ __ESIMD_API std::enable_if_t<std::is_integral<T0>::value &&
                                  std::is_integral<U>::value,
                              __ESIMD_NS::simd<T0, SZ>>
 rol(__ESIMD_NS::simd<T1, SZ> src0, U src1) {
-  using ComputationTy =
-      __ESIMD_DNS::computation_type_t<decltype(src0), int32_t>;
-  ComputationTy Src0 = src0;
-  ComputationTy Src1 = src1;
-  return esimd::rol<T0>(Src0, Src1);
+  constexpr auto nrOfDigits = std::numeric_limits<T1>::digits;
+  __ESIMD_NS::simd<T1, SZ> Src1 = src1 % nrOfDigits;
```

Comment:
Division is extremely expensive operation. 
Normally, compiler should replace _reminder_ operation with masked-AND when the op2 is known to be a power of 2.
Can you check if that optimization happens in case of simd?

Also, even if suppose "SimdA % 16" is optimized with "SimdA & 0x0F", I still would prefer to avoid that operation if possible, and have only HW operation ROL instead of ROL + const-creation + AND.

@kbobrovs - what do you think?

Reply:
> The idea was to optimize the performance by avoiding rotating more than actual number of bits in the 1st argument

It should map to a hardware instruction, which takes same amount of time for shift count equal to K and (64+K)

Reply:
Yes, I think we should avoid unnecessary `%` `/`.